src/main/java/com/kainos/ea/controller/WebService.java

```diff
@@ -38,6 +61,125 @@ public Response getJobRoles()
 
     }
 
+    @GET
+    @Path("/job-roles/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobRoleById(@PathParam("id") int id){
+        try {
+            return Response.status(HttpStatus.OK_200).entity(rolesService.getRoleById(id)).build();
+        } catch (Exception | DatabaseConnectionException  e){
+            return Response.status(HttpStatus.INTERNAL_SERVER_ERROR_500, e.getMessage()).build();
+        } catch (JobRoleDoesNotExistException e){
```

Comment:
this needs to go above the other catch as the other catch catches generic exception class

Reply:
done